src/Graph.ts

```diff
@@ -396,7 +421,7 @@ export default class Graph {
 				chunkList.forEach(chunk => {
 					// generate the imports and exports for the output chunk file
 					if (chunk.entryModule) {
-						const entryName = generateChunkName(chunk.entryModule.id, chunkNames, true);
+						const entryName = generateChunkName(chunk.entryModule.id, chunkNames, true, inputRelativeDir);
```

Comment:
Is this a mandatory option then? Is there not a way to deduce the lowest base by default if this is not provided?

Reply:
I was able to rely on the package [commondir](https://github.com/substack/node-commondir) to calculate given the entry modules. It is used in a subsequent feature request [here](https://github.com/kellyselden/rollup/commit/90ded1adb48521c7b5a07d63a83bcf6d9c2bed5b#diff-ef9931d02c30e05453c44ad0b6281929R8) to make it something other than common base path, but I will move the option into that branch instead.

Reply:
Since `commondir` is not widely maintained and a small piece of code, do you think we could just inline the implementation?

Reply:
I could do that. Are you sure I should just copy paste? I would argue that we depend on it until we need to fix something, then copy. @lukastaegert care to tie-break?

Reply:
I must admit I did not read everything here. Looking at commondir, it relies on nodes `path.resolve` which we do not have available in a browser build—even though we have our own polyfill for that (damn, we REALLY need a test for this. You should see a warning about "path" being external when building rollup with this setup—this is always BAD). Thus I fear this needs to go into our own code base somehow.

However I am happy for every bit of functionality we can extract from Graphs.ts and put into other files and for me, this is a clear candidate for the "utils" folder.

Reply:
ported into the utils folder.